File: selfstudy/golang/students_counthigh.go
/*
task 217:
Students - CountHigh
You need to count how many values are greater than 50 in student test scores
*/

package main

import (
	"fmt"
	"reflect"
)

func Count_above_fiftyStudents(scores []int) int {
	count := 0
	for _, v := range scores {
		if v > 50 { count++ }
	}
	return count
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, Count_above_fiftyStudents([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Count_above_fiftyStudents([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Count_above_fiftyStudents([]int{1, 2, 3})},
		{[]int{100, 200}, Count_above_fiftyStudents([]int{100, 200})},
		{[]int{}, Count_above_fiftyStudents([]int{})},
	}
	
	for idx, t := range tests {
		res := Count_above_fiftyStudents(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done

File: selfstudy/js/students_counthigh.js
/*
task 17:
Students - CountHigh
You need to count how many values are greater than 50 in student test scores
*/

function count_above_fifty_students(scores) {
  return scores.filter(x => x > 50).length;
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: count_above_fifty_students([10, 20, 30]) },
  { input: [50, 60, 70], expected: count_above_fifty_students([50, 60, 70]) },
  { input: [1, 2, 3], expected: count_above_fifty_students([1, 2, 3]) },
  { input: [100, 200], expected: count_above_fifty_students([100, 200]) },
  { input: [], expected: count_above_fifty_students([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('count_above_fifty_students', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(count_above_fifty_students(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, count_above_fifty_students(t.input) === t.expected || JSON.stringify(count_above_fifty_students(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// works i think
